libraries/AP_InertialSensor/AP_InertialSensor_ExternalAHRS.cpp

```diff
@@ -0,0 +1,52 @@
+#include <AP_HAL/AP_HAL.h>
+#include "AP_InertialSensor_ExternalAHRS.h"
+#include <AP_ExternalAHRS/AP_ExternalAHRS.h>
+
+#if HAL_EXTERNAL_AHRS_ENABLED
+
+const extern AP_HAL::HAL& hal;
+
+AP_InertialSensor_ExternalAHRS::AP_InertialSensor_ExternalAHRS(AP_InertialSensor &imu, uint8_t _serial_port) :
+    AP_InertialSensor_Backend(imu),
+    serial_port(_serial_port)
+{
+}
+
+void AP_InertialSensor_ExternalAHRS::handle_external(const AP_ExternalAHRS::ins_data_message_t &pkt)
+{
+    Vector3f accel{pkt.accel[0], pkt.accel[1], pkt.accel[2]};
+    Vector3f gyro{pkt.gyro[0], pkt.gyro[1], pkt.gyro[2]};
+
+    _rotate_and_correct_accel(accel_instance, accel);
+    _notify_new_accel_raw_sample(accel_instance, accel, AP_HAL::micros64());
+
+    _publish_temperature(accel_instance, pkt.temperature);
```

Comment:
I wonder if we use this temperature in our reporting anywhere?  It's obviously nothing to do with the autopilot temperature in this case....

Reply:
we do log it in IMU msg